streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/testing/EventLoggableOutputCollector.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2017 Hortonworks.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *   http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package com.hortonworks.streamline.streams.runtime.storm.testing;
+
+import com.hortonworks.streamline.streams.StreamlineEvent;
+
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public interface EventLoggableOutputCollector {
```

Comment:
Yes the interface is intended to be used like a mix-up / trait. Two methods are needed from both EventLoggingOutputCollector and EventLoggingSpoutOutputCollector, and they're having different class hierarchy so I thought easiest way to add methods to both of them is using trait-like interface.

I don't have strong opinion of keeping this as it is. Just changed to what you suggested. That looks like better that they no longer expose component name and event logger to the public.